src/execution-context.js

```diff
@@ -1,227 +1,4 @@
-        }
-      })
-    }
-  }
-
-  this.internalWeb3 = function () {
-    return web3
-  }
-
-  this.blankWeb3 = function () {
-    return blankWeb3
-  }
-
-  this.vm = function () {
-    return vm
-  }
-
-  this.setContext = function (context, endPointUrl, confirmCb, infoCb) {
-    executionContext = context
-    this.executionContextChange(context, endPointUrl, confirmCb, infoCb)
-  }
-
-  this.executionContextChange = function (context, endPointUrl, confirmCb, infoCb, cb) {
-    if (!cb) cb = () => {}
-
-    if (context === 'vm') {
-      executionContext = context
-      vm.stateManager.revert(function () {
-        vm.stateManager.checkpoint()
-      })
-      self.event.trigger('contextChanged', ['vm'])
-      return cb()
-    }
-
-    if (context === 'injected') {
-      if (injectedProvider === undefined) {
-        var alertMsg = 'No injected Web3 provider found. '
-        alertMsg += 'Make sure your provider (e.g. MetaMask) is active and running '
-        alertMsg += '(when recently activated you may have to reload the page).'
-        infoCb(alertMsg)
-        return cb()
-      } else {
-        executionContext = context
-        web3.setProvider(injectedProvider)
-        self.event.trigger('contextChanged', ['injected'])
-        return cb()
-      }
-    }
-
-    if (context === 'web3') {
-      confirmCb(cb)
-    }
-  }
-
-  this.currentblockGasLimit = function () {
-    return this.blockGasLimit
-  }
-
-  this.blockGasLimitDefault = 4300000
-  this.blockGasLimit = this.blockGasLimitDefault
-  setInterval(() => {
-    if (this.getProvider() !== 'vm') {
-      web3.eth.getBlock('latest', (err, block) => {
-        if (!err) {
-          // we can't use the blockGasLimit cause the next blocks could have a lower limit : https://github.com/ethereum/remix/issues/506
-          this.blockGasLimit = (block && block.gasLimit) ? Math.floor(block.gasLimit - (5 * block.gasLimit) / 1024) : this.blockGasLimitDefault
-        } else {
-          this.blockGasLimit = this.blockGasLimitDefault
-        }
-      })
-    }
-  }, 15000)
-
-  // TODO: not used here anymore and needs to be moved
-  function setProviderFromEndpoint (endpoint, context, cb) {
-    var oldProvider = web3.currentProvider
-
-    if (endpoint === 'ipc') {
-      web3.setProvider(new web3.providers.IpcProvider())
-    } else {
-      web3.setProvider(new web3.providers.HttpProvider(endpoint))
-    }
-    if (web3.isConnected()) {
-      executionContext = context
-      self.event.trigger('contextChanged', ['web3'])
-      self.event.trigger('web3EndpointChanged')
-      cb()
-    } else {
-      web3.setProvider(oldProvider)
-      var alertMsg = 'Not possible to connect to the Web3 provider. '
-      alertMsg += 'Make sure the provider is running and a connection is open (via IPC or RPC).'
-      cb(alertMsg)
-    }
-  }
-  this.setProviderFromEndpoint = setProviderFromEndpoint
-}
+var executionContext = remixLib.execution.executionContext
```

Comment:
we need to fix the code in `execution-context` before merging that.
that's a change I made last week to fix tape tests execution...